org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JobHelpers.java

```diff
@@ -135,11 +137,19 @@ private static boolean flushProcessingQueues(IJobManager jobManager, IProgressMo
 	}
 
 	private static void waitForBuildJobs() {
-		waitForJobs(BuildJobMatcher.INSTANCE, 300000);
+		waitForJobs(BuildJobMatcher.INSTANCE, MAX_TIME_MILLIS);
 	}
 
 	public static void waitForInitializeJobs() {
-		waitForJobs(InitializeJobMatcher.INSTANCE, 300000);
+		waitForJobs(InitializeJobMatcher.INSTANCE, MAX_TIME_MILLIS);
+	}
+
+	public static void waitForDownloadSourcesJobs() {
+		waitForJobs(DownloadSourcesJobMatcher.INSTANCE, WAIT_FOR_DOWNLOAD_SOURCE_JOB_MILLIS);
+	}
+
+	public static void waitForDownloadSourcesJobsTest() {
```

Comment:
no, this class shouldn't have test specific methods. I'd rather prefer you added 
public static void waitForDownloadSourcesJobs(int maxWaitMillis)